esx_service/utils/auth_data.py

```diff
@@ -0,0 +1,568 @@
+# VM based authorization for docker volumes
+# Copyright 2016 VMware, Inc. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License
+
+""" VM based authorization for docker volumes and tenant management.
+
+"""
+
+import sqlite3
+import uuid
+import os
+import vmdk_utils
+import vmdk_ops
+import logging
+import auth_data_const
+ 
+class DbConnectionError(Exception):
+    """ An exception thrown when connection to a sqlite database fails. """
+
+    def __init__(self, db_path):
+        self.db_path = db_path
+
+    def __str__(self):
+        return "DB connection error %s" % self.db_path
+
+class DockerVolumeTenant:
+    """ This class abstracts the operations to manage a DockerVolumeTenant.
+
+    The interfaces it provides include:
+    - add VMs to tenant
+    - revmove VMs from tenant
+    - change tenant name and description
+    - set datastore and privileges for a tenant
+
+    """
+
+    def __init__(self, name, description, default_datastore, default_privileges,
+                    vms, privileges, id=None):
+            """ Constuct a DockerVOlumeTenant object. """
+            self.name = name
+            self.description = description
+            self.default_datastore = default_datastore
+            self.default_privileges = default_privileges
+            self.vms = vms
+            self.privileges = privileges
+            if not id:
+                self.id = str(uuid.uuid4())
+            else:
+                self.id = id
+        
+    def add_vms(self, conn, vms):
```

Comment:
We probably want to return errors for this and other operations, since they'll need to be sent back to the user in production.